File: java-automation/src/main/java/com/automation/data/TestData.java
package com.automation.data;

public class TestData {
    public static class Users {
        public static final String STANDARD_USERNAME = "standard_user";
        public static final String STANDARD_PASSWORD = "secret_sauce";
        public static final String LOCKED_USERNAME = "locked_out_user";
        public static final String LOCKED_PASSWORD = "secret_sauce";
        public static final String INVALID_USERNAME = "invalid_user";
        public static final String INVALID_PASSWORD = "wrong_password";
    }

    public static class Urls {
        public static final String BASE = "https://www.saucedemo.com";
        public static final String INVENTORY = "https://www.saucedemo.com/inventory.html";
        public static final String CART = "https://www.saucedemo.com/cart.html";
        public static final String CHECKOUT_STEP_1 = "https://www.saucedemo.com/checkout-step-one.html";
        public static final String CHECKOUT_STEP_2 = "https://www.saucedemo.com/checkout-step-two.html";
        public static final String CHECKOUT_COMPLETE = "https://www.saucedemo.com/checkout-complete.html";
    }

    public static class ErrorMessages {
        public static final String LOCKED_OUT = "Epic sadface: Sorry, this user has been locked out.";
        public static final String INVALID_CREDENTIALS = "Epic sadface: Username and password do not match any user in this service";
        public static final String MISSING_USERNAME = "Epic sadface: Username is required";
        public static final String MISSING_PASSWORD = "Epic sadface: Password is required";
    }

    public static class CustomerData {
        public static final String FIRST_NAME = "Test";
        public static final String LAST_NAME = "User";
        public static final String POSTAL_CODE = "12345";
    }
}

File: reqnroll-automation/playwright/data/TestData.cs
namespace ReqnrollPlaywright.Fixtures;

public static class Users
{
    public const string StandardUsername = "standard_user";
    public const string StandardPassword = "secret_sauce";
    public const string LockedUsername = "locked_out_user";
    public const string LockedPassword = "secret_sauce";
    public const string PerformanceGlitchUsername = "performance_glitch_user";
    public const string PerformanceGlitchPassword = "secret_sauce";
    public const string InvalidUsername = "invalid_user";
    public const string InvalidPassword = "wrong_password";
}

public static class Urls
{
    public const string Base = "https://www.saucedemo.com";
    public const string Inventory = "https://www.saucedemo.com/inventory.html";
    public const string Cart = "https://www.saucedemo.com/cart.html";
    public const string CheckoutStep1 = "https://www.saucedemo.com/checkout-step-one.html";
    public const string CheckoutStep2 = "https://www.saucedemo.com/checkout-step-two.html";
    public const string CheckoutComplete = "https://www.saucedemo.com/checkout-complete.html";
}

public static class ErrorMessages
{
    public const string LockedOut = "Epic sadface: Sorry, this user has been locked out.";
    public const string InvalidCredentials = "Epic sadface: Username and password do not match any user in this service";
    public const string MissingUsername = "Epic sadface: Username is required";
    public const string MissingPassword = "Epic sadface: Password is required";
    public const string MissingFirstName = "Error: First Name is required";
    public const string MissingLastName = "Error: Last Name is required";
    public const string MissingPostalCode = "Error: Postal Code is required";
}

public static class CustomerData
{
    public const string FirstName = "Devbrat";
    public const string LastName = "Verma";
    public const string PostalCode = "110001";
}
